### PR TITLE
List Renderer

### PR DESCRIPTION
**Pull request by maaslalani**
## Rendering Lists

Lip Gloss ships with a list rendering sub-package.

```go
import "github.com/charmbracelet/lipgloss/list"
```

Define a new list.

```go
l := list.New("A", "B", "C")
```

Print the list.

```go
fmt.Println(l)

// • A
// • B
// • C
```

### Customization

Lists can be customized via their enumeration function as well as using
`lipgloss.Style`s.

```go
enumeratorStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("99")).MarginRight(1)
itemStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("212")).MarginRight(1)

l := list.New(
  "Glossier",
  "Claire’s Boutique",
  "Nyx",
  "Mac",
  "Milk",
).
  Enumerator(list.Roman).
  EnumeratorStyle(enumeratorStyle).
  ItemStyle(itemStyle)
```

Print the list.

<p align="center">
List example
</p>

In addition to the predefined enumerators (`Arabic`, `Alphabet`, `Roman`, `Bullet`, `Tree`),
you may also define your own custom enumerator:

```go
var DuckDuckGooseEnumerator Enumerator = func(l *List, i int) string {
    if l.At(i) == "Goose" {
        return "Honk →"
    }
    return ""
}
```

Use it in a list:

```go
l := list.New("Duck", "Duck", "Duck", "Duck", "Goose", "Duck", "Duck")
l.Enumerator(DuckDuckGooseEnumerator)
```

Print the list:

<p align="center">
image
</p>

### Building

If you need, you can also build lists incrementally:

```go
l := list.New()

for i := 0; i < repeat; i++ {
    l.Item("Lip Gloss")
}
```

### CODE REVIEW

**Review by caarlos0 — APPROVED** [review]
this looks great!

File: list/enumerations.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	return fmt.Sprintf("%d.", i+1)
+}
+
+// Roman is the enumeration for roman numerals listing.
+//
+// /   I. Foo
+// /  II. Bar
+// / III. Baz
+// /  IV. Qux.
+func Roman(_ *List, i int) string {
```
Comment (muesli) [inline list/enumerations.go:46]: Love it.

**Comment by muesli** [comment]
Similar to our `table` implementation, I think it would be desirable to support custom data models.

**Comment by testinfected** [comment]
Hello guys, 

I'm in need for this. Can we expect a merge to master anytime soon?

Thanks

**Comment by maaslalani** [comment]
* Need to add `ItemStyleFunc` and `EnumeratorStyleFunc` as well to allow more customization of the list.

* `ItemStyle` and `EnumeratorStyle` should be aliases for these function.

**Comment by caarlos0** [comment]
> * Need to add `ItemStyleFunc` and `EnumeratorStyleFunc` as well to allow more customization of the list.
> * `ItemStyle` and `EnumeratorStyle` should be aliases for these function.

implemented in #264

**Comment by caarlos0** [comment]
superseeded by #264

**Comment by caarlos0** [comment]
closing in favor of https://github.com/charmbracelet/lipgloss/pull/264

**Pull request closed without merging**